index.js

```diff
@@ -1,260 +1,216 @@
-		this.log('info', 'Attempting to reconnect to switcher')
-		this.disconnect()
-
-		if (retry_immediately) {
-			this.connect()
-		} else {
-			this.connect_timeout(this.RECONNECT_TIMEOUT)
+	initWebSocket() {
+		if (this.reconnect_timer) {
+			clearTimeout(this.reconnect_timer)
+			this.reconnect_timer = null
 		}
-	}
 
-	/**
-	 * Try to commect again after timeout
-	 * @param {Int} timeout - Timeout to try reconnection (seconds)
-	 */
-	connect_timeout(timeout) {
-		if (this.reconnecting) {
+		const url = 'ws://' + this.config.host + ':8621'
+		this.log('info', 'Connecting: ' + url)
+		if (!url || url.match(new RegExp(this.wsRegex)) === null) {
+			this.updateStatus(InstanceStatus.BadConfig, `WS URL is not defined or invalid`)
 			return
 		}
-		this.log('info', `Attempting to reconnect in ${timeout} seconds.`)
-		this.reconnecting = setTimeout(this.connect.bind(this, true), timeout * 1000)
-	}
 
-	/**
-	 * Send a command to the switcher
-	 * @param {string} command - The command to send
-	 */
-	sendCommand(command) {
-		if (!this.socket || !this.socket.connected) {
-			this.log('warn', 'Switcher not connected')
-			this.reconnect.bind(this, true)
-			return
-		}
-		this.debug(`Sending command: ${command}`)
-		this.socket.sendUTF(command)
-	}
+		this.updateStatus(InstanceStatus.Connecting)
 
-	/**
-	 * Disconccect from device
-	 */
-	disconnect() {
-		this.log('info', 'Disconnecting from switcher')
-		this.status(this.STATUS_UNKNOWN)
-		if (this.socket && this.socket.connected) {
-			this.socket.close()
+		if (this.ws) {
+			this.ws.close(1000)
+			delete this.ws
+			this.isConnected = false
 		}
-	}
+		this.ws = new WebSocket(url, {'origin': this.config.host})
 
-	/**
-	 * Ends session if connected
-	 */
-	destroy() {
-		this.disconnect()
+		this.ws.on('open', () => {
+			this.updateStatus(InstanceStatus.Ok)
+			this.isConnected = true;
+			this.log('debug', `Connection opened`)
+			this.ws.send(this.getCommandForAction(this.config.model, 'get_state', null))
+		})
+		this.ws.on('close', (code) => {
+			this.log('debug', `Connection closed with code ${code}`)
+			this.updateStatus(InstanceStatus.Disconnected, `Connection closed with code ${code}`)
+			this.isConnected = false
+			this.maybeReconnect()
+		})
+
+		this.ws.on('message', this.messageReceivedFromWebSocket.bind(this))
+
+		this.ws.on('error', (data) => {
+			this.log('error', `WebSocket error: ${data.stack}`)
+		})
+	}
+	
+	messageReceivedFromWebSocket(msg) {
+		//if (this.config.debug_messages) {
+		//}
+		const message = msg.toString('utf8');
+		this.log('info', "Received: " + message)
+		message
+			.split(',')
+			.map((item) => item.trim())
+			.forEach((item) => {
+				//this.log('warn','Data recieved: ' + item)
+				if (item.match('^[A-Za-z0-9_:]*$') !== null) {
+					let result = this.parseVariable(item)
+					if (result !== null) {
+						let jsonStr = '{"' + result[0] + '":"' + result[1] + '"}'
+						this.setVariableValues(JSON.parse(jsonStr))
```

Comment:
Is there a reason for this complicated and time consuming procedure?
Why not 
```suggestion
						this.setVariableValues({[result[0]]: result[1]})
```